File: app/src/androidTest/java/com/example/application1/HomeActivityTests.java
package com.example.application1;

import static androidx.test.espresso.matcher.ViewMatchers.withId;

import androidx.annotation.ContentView;
import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.rules.ActivityScenarioRule;

import com.example.application1.app.HomeActivity;

public class HomeActivityTests {
    var mActivityScenerioRule = ActivityScenarioRule(HomeActivity::class.java)

    fun logicTest(){
        onView(withId(R.id.etUi))
                .perform(typeText)
    }
}

File: app/src/androidTest/java/com/example/application1/app/HomeActivityTests.kt
package com.example.application1.app

import androidx.core.content.MimeTypeFilter.matches
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.action.ViewActions.closeSoftKeyboard
import androidx.test.espresso.action.ViewActions.typeText
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import com.example.application1.R
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import java.util.regex.Pattern.matches
import kotlin.reflect.typeOf


@LargeTest
@RunWith(AndroidJUnit4::class)
class HomeActivityTests {

    @Rule
    @JvmField
    var mActivityScenarioRule = ActivityScenarioRule(HomeActivity::class.java)

    @Test
    fun loginTest() {
        onView(withId(R.id.etUi))
            .perform(typeText("hello"), closeSoftKeyboard())

        onView(withId(R.id.btnUi)).perform(click())
            .check(matches(withText("hello")));
    }

}